### DISCUSSION TITLE
Unable to configure URL prefix with Apache Alias

### DISCUSSION THREAD

**Discussion by rv2931** [apache/airflow#28225] [Q&A]
Hello
I'm trying to use Airflow docker installation on a serveur but accessing to the instance with a apache alias suffix but I don't manage to get it right
I've read the doc : https://airflow.apache.org/docs/apache-airflow/stable/howto/run-behind-proxy.html
The exemple of web server configuration is for Nginx but I'm using apache
Here is my apache configuration:
```
# /etc/apache2/sites-enabled/airflow.conf
ProxyPass /my_prefix/airflow http://127.0.0.1:28001/
ProxyPassReverse /my_prefix/airflow http://127.0.0.1:28001/
ProxyPreserveHost On
```

So my container expose the 8080 guest port on 28001 host port

Here is my airflow configuration (airflow conf list):
```
# /opt/airflow/airflow.cfg
base_url = http://localhost:8080/my_prefix/airflow
enable_proxy_fix = True
proxy_fix_x_for = 1 (default)
proxy_fix_x_proto = 1 (default)
proxy_fix_x_host = 1 (default)
proxy_fix_x_port = 1 (default)
proxy_fix_x_prefix = 1 (default)
```

But I get 404 or 302 answer, but not home page or "Apache Airflow is not at this location" answer when trying https://myhost/my_prefix/airflow or https://myhost:28001
Do I missed something ? I think it is coming from apache2 conf but which parameter is missing ?
Thank you for reading and for your answers and questions
RV

**Comment by potiuk** [comment]
I am not profficient in Apache Server configuration, so can't give you a "fish" but I can give you a "fishing rod". I think it hs something to do with URL rewrites - and this is not an airlfow issue - but generic issue that you have when you want to configure Apache server and you will likely find some good guides by googling. 

I'd suggest you to take a look at the logs of your apache server logs and Airlfow webserver logs and see what happens when you connect. That should give you enough understanding of where the problem is an what needs to be handled (this is what I did in the past when I had similar problems).

**Comment by rv2931** [comment]
it's clearly coming from apache configuration... but whichc parameter is missing
On the same serveur I tried nginx conf and apache conf. It's working perfectly on http://airflow.mydomain.com/airflow with nginx with simple bloc:
```
server {
  listen 80;
  server_name airflow.mydomain.com;

  location /airflow/ {
      proxy_pass http://localhost:8080;
  }
}
```
On apache side, I try : 
```
<VirtualHost *:80>
        ServerName airflow.mydomain.com
        ProxyPass /airflow http://localhost:8080
        ProxyPassReverse /airflow http://localhost:8080
        ProxyPreserveHost On
</VirtualHost>
```
and the response is : 
in apache I have no error in errors.log, just a 404 in other_vhost_access.log:
`airflow.mydomain.com:80 192.168.0.164 - - [08/Dec/2022:21:44:13 +0100] "GET /airflow/ HTTP/1.1" 404`
I can imagine that the 404 is return by airflow and not by apache

It looks than the apache equivalent of nginx (that I don't raly know) is not as simple than nginx...

**Reply by potiuk** [reply]
Maybe you have not realoaded the config :) hard ot say- not an apache conf expert :)
